### ISSUE TITLE
Unexpected diagnostic in `with` expression with indexer on a struct

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#81666]
Although this is an error scenario, the first diagnostic is unexpected.
```
    [Fact]
    public void TODO2()
    {
        var src = """
var c = new S() with { [0] = 1 };

public struct S
{
    public int this[int i] { set { } }
}
""";

        CreateCompilation(src).VerifyEmitDiagnostics(
            // (1,24): error CS0131: The left-hand side of an assignment must be a variable, property or indexer
            // var c = new S() with { [0] = 1 };
            Diagnostic(ErrorCode.ERR_AssgLvalueExpected, "[0]").WithLocation(1, 24),
            // (1,24): error CS0747: Invalid initializer member declarator
            // var c = new S() with { [0] = 1 };
            Diagnostic(ErrorCode.ERR_InvalidInitializerElementInitializer, "[0] = 1").WithLocation(1, 24));
    }
```

**Comment by AlekseyTs** [comment]
Is the second error expected though?
